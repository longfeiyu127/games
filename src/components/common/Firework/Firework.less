@particles: 100;
@width: 500;

@count: 60;

@var: `Math.random() `;

@box-shadow: 0 0 #fff;

@0px: 0px;

.loop(@i) when (@i < @count) {
  // @x: @var * @width;
  // @y: @var * @width;
  box-shadow+: @var* @width - @width / 2 + @0px @var* @width - @width / 1.2 + @0px hsl(@var*360, 100%, 50%);
  .loop(@i+1);
}

.loop1(@i) when (@i < @count) {
  box-shadow+: @box-shadow;
  .loop1(@i+1);
}

.c-firework {
  position: absolute;
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
  .firework {
    position: absolute;
    top: 50%;
    width: 80%;
    margin-top: -150px;
    height: 300px;
    .light {
      animation: 1s bang ease-out infinite backwards, 1s gravity ease-in infinite backwards,
        5s position linear infinite backwards;
      border-radius: 50%;
      .loop1(1);
      height: 5px;
      position: absolute;
      width: 5px;
      &.after {
        // animation-delay: 10s, 10s, 10s;
        animation-duration: 1.25s, 1.25s, 6.25s;
      }
    }
  }
}

@keyframes bang {
  to {
    .loop(1);
  }
}

@keyframes gravity {
  to {
    opacity: 0;
    transform: translateY(200px);
  }
}

@keyframes position {
  0%,
  19.9% {
    margin-left: 40%;
    margin-top: 10%;
  }

  20%,
  39.9% {
    margin-left: 30%;
    margin-top: 40%;
  }

  40%,
  59.9% {
    margin-left: 70%;
    margin-top: 20%;
  }

  60%,
  79.9% {
    margin-left: 20%;
    margin-top: 30%;
  }

  80%,
  99.9% {
    margin-left: 80%;
    margin-top: 30%;
  }
}
