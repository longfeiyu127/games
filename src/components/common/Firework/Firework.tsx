import React from 'react'
import './Firework.less'

const Firework = () => (
  <div className="c-firework">
    <div className="firework">
      <div className="light before" />
      <div className="light after" />
    </div>
  </div>
)

export default Firework
