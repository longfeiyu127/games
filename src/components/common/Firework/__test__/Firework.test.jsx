import React from 'react'
import renderer from 'react-test-renderer'
import Firework from '../Firework.tsx'

describe('<Firework />', () => {
  it('component common/Firework render', () => {
    const Dom = renderer.create(<Firework />).toJSON()
    expect(Dom).toMatchSnapshot()
  })
})
