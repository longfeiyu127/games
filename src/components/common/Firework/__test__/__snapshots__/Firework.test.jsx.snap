// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Firework /> component common/Firework render 1`] = `
<div
  className="c-firework"
>
  <div
    className="firework"
  >
    <div
      className="light before"
    />
    <div
      className="light after"
    />
  </div>
</div>
`;
